### ISSUE TITLE
Do not emit diagnostics about unused arguments when dealing with protocol methods

### ISSUE THREAD

**Issue by rwols** [microsoft/pyright#894]
**Describe the bug**
When you define a mypy `Protocol`, pyright will point out that the arguments of a method are unused. This is intended, seeing as [I'm defining a protocol](https://mypy.readthedocs.io/en/stable/protocols.html#simple-user-defined-protocols).

**To Reproduce**
Smallest example I could come up with:
```python
from typing_extensions import Protocol


class MyProtocol(Protocol):

    def foo(self, x: int) -> None:
        ...
```

**Expected behavior**
No diagnostics as I'm defining a protocol.

**Screenshots or Code**
![Screenshot from 2020-07-30 21-20-57](https://user-images.githubusercontent.com/2431823/88965118-9425cd00-d2aa-11ea-8669-0e5ea89eaa2f.png)

**VS Code extension or command-line**
N/A

**Additional context**
It might be safe to not report such diagnostics whenever the body of the function/method is `...`.

**Comment by jakebailey** [comment]
In VS Code, this message is not shown at all as it can handle the `Unnecessary` diagnostic tag, which indicates that the editor should be styling the code differently, e.g. graying it out. If the client doesn't support that case, then it's going to show "real" diagnostics in places we aren't expecting the user to actually see anything.

Perhaps this should be conditional based on a capability, but this behavior is in the spec.

**Comment by erictraut** [comment]
Yes, as Jake said, this is intended behavior. Pyright is emitting these with an "unnecessary" diagnostic tag, so a client should not display them as real diagnostics but instead display the corresponding text as gray.

**Comment by rwols** [comment]
Interesting! Thanks for pointing this out.

**Comment by rwols** [comment]
Made an issue about it here https://github.com/sublimelsp/LSP/issues/1227
